File: ruby/aa/largest_cont_sub_sum.rb
def largest_cont_sub_sum(nums)
  max_sum = 0
  curr_sum = 0

  nums.each do |num|
    curr_sum += num
    max_sum = curr_sum if curr_sum > max_sum
    curr_sum = 0 if curr_sum < 0
  end

  max_sum
end

p largest_cont_sub_sum([4,5,-3,2,-6])

File: js/aa/largestContSubSum.js
function largestContSubSum(nums) {
  let largest = 0;
  let currSum = 0;
  for (let i = 0; i < nums.length; i++) {
    if (currSum + nums[i] > 0) {
      currSum += nums[i];
      if (currSum > largest) {
        largest = currSum;
      }
    } else {
      currSum = 0;
    }
  }

  return largest;
}

let sum = largestContSubSum([2,3,-3,5])
console.log(sum);
